src/x11/xf86Wacom.c

```diff
@@ -525,22 +531,39 @@ void wcmInitAxis(WacomDevicePtr priv, enum WacomAxisType type,
 			index = 4;
 			label = XIGetKnownProperty(AXIS_LABEL_PROP_ABS_THROTTLE);
 			break;
+		case WACOM_AXIS_SCROLL_X:
+			index = 5;
+			break;
+		case WACOM_AXIS_SCROLL_Y:
+			index = 6;
+			break;
 		case WACOM_AXIS_WHEEL:
 		case WACOM_AXIS_RING:
-			index = 5;
+			index = 7;
 			label = XIGetKnownProperty(AXIS_LABEL_PROP_ABS_WHEEL);
 			break;
 		case WACOM_AXIS_RING2:
-			index = 6;
+			index = 8;
 			break;
+
 		default:
 			abort();
 	}
 
-	InitValuatorAxisStruct(pInfo->dev, index,
-	                       label,
-	                       min, max, res, min_res, max_res,
-	                       Absolute);
+
+	if (type == WACOM_AXIS_SCROLL_X) {
+		DBG(0, priv, "Initting x scroll axis: (pInfo->dev, %d, %x, 120, 0)\n", index, SCROLL_TYPE_HORIZONTAL);
+		SetScrollValuator(pInfo->dev, index, SCROLL_TYPE_HORIZONTAL, 120, 0);
+	} else if (type == WACOM_AXIS_SCROLL_Y) {
+		DBG(0, priv, "Initting y scroll axis: (pInfo->dev, %d, %x, 120, 0)\n", index, SCROLL_TYPE_VERTICAL);
+		SetScrollValuator(pInfo->dev, index, SCROLL_TYPE_VERTICAL, 120, 0);
```

Comment:
note: `SetScrollValuator` is **in addition** to `InitValuatorAxisStruct`. It's like an extra flag of "use this valuator for scrolling"

Reply:
Is it? I based my code on this: https://github.com/freedesktop/xorg-xf86-input-libinput/blob/62f267a952208005bf378c554fa84f4dc2489b35/src/xf86libinput.c#L938, which doesn't call `InitValuatorAxisStruct` on the scroll ones. Is that an error on their side that is silently ignored? Is there any documentation or reference on this?

Reply:
Is it? I based my code on this: https://github.com/freedesktop/xorg-xf86-input-libinput/blob/62f267a952208005bf378c554fa84f4dc2489b35/src/xf86libinput.c#L938, which doesn't call `InitValuatorAxisStruct` on the scroll ones. Is that an error on their side that is silently ignored? Is there any documentation or reference on this?

Reply:
hmm, that is actually a bug in that driver. it just happened to not be a problem because the defaults used for valuators happen to match what we need and the valuator semi-exists because we already initialized the number of valuators earlier.

Either way, it's not technically correct, just happened to work.